pulp_docker/app/tasks/recursive_remove.py

```diff
@@ -0,0 +1,98 @@
+from pulpcore.plugin.models import Content, Repository, RepositoryVersion
+
+from pulp_docker.app.models import Blob, Manifest, MEDIA_TYPE, Tag
+
+
+def recursive_remove_content(repository_pk, content_units):
+    """
+    Create a new repository version by recursively removing content.
+
+    For each unit that is specified, we also need to remove related content,
+    unless that content is also related to content that will remain in the
+    repository. For example, if a manifest-list is specified, we need to remove
+    all referenced manifests unless those manifests are referenced by a
+    manifest-list that will stay in the repository.
+
+    Args:
+        repository_pk (int): The primary key for a Repository for which a new Repository Version
+            should be created.
+        content_units (list): List of PKs for :class:`~pulpcore.app.models.Content` that
+            should be removed from the Repository.
+    """
+    repository = Repository.objects.get(pk=repository_pk)
+    latest_version = RepositoryVersion.latest(repository)
+    latest_content = latest_version.content.all() if latest_version else Content.objects.none()
+
+    tags_to_remove = Tag.objects.filter(
+        pk__in=content_units,
```

Comment:
I am wondering whether we need to ensure that `tags_to_remove` contains tags that actually are in `latest_content`. Of course, tags that are not in the latest repo version won't be removed, but such a tag may reference a manifest/manifest list that is in the latest repo version. I think it is surprising if such a manifest is removed although the tag referencing it isn't even part of the repo version (and is not removed).

Same reasoning applies to manifest (lists).



If I understand the code below correctly, such a manifest may be removed